src/lib/notifications.ts

```diff
@@ -0,0 +1,90 @@
+import { AppError } from 'src/lib/errors'
+import { randomBytes } from 'crypto'
+
+interface NotificationMessage {
+  title: string
+  message?: string
+}
+
+/**
+ * A notification is either dismissible or not
+ * if it is dismissible it can also optionally have
+ * - a label (for the dismiss button)
+ * - a timeout (to auto-dismiss)
+ * - a callback
+ */
+
+type NotificationDismiss = {
+  dismiss?: boolean | {
+    label?: string,
+    timeout?: number,
+    onDismiss?: (...args: any) => void
+  },
+}
+
+/**
+ * A notification may have a call to action button,
+ * with a label and an onInteract callback
+ */
+type NotificationInteract = {
+  interact?: {
```

Comment:
Just for consistency, perhaps this should be defined like `NotificationDismiss`, i.e. 
```{
interact?: boolean | {
...
}
}```

Or the other way around.

Reply:
For `interact` it doesn't really make sense to just have `interact: true` because if you specify a possible interaction you *must* specify some callback and label for it, or otherwise it is useless